OpenImis.ePayment/Escape/Payment/Data/GepgUtility.cs

```diff
@@ -206,245 +187,157 @@ public String CreateGePGCancelPaymentRequest(IConfiguration Configuration, int P
+                    Console.WriteLine(ex.ToString());
                 }
-            }
 
-            return outString;
+                return Encoding.UTF8.GetString(sb.ToArray());
+            }
         }
 
         public string SerializeClean(object bill, Type type)
         {
-            XmlSerializer xs = null;
-            //These are the objects that will free us from extraneous markup.
-            XmlWriterSettings settings = null;
-            XmlSerializerNamespaces ns = null;
-
-            //We use a XmlWriter instead of a StringWriter.
-            XmlWriter xw = null;
-
-            String outString = String.Empty;
-
-            try
-            {
-                //To get rid of the xml declaration we create an 
-                //XmlWriterSettings object and tell it to OmitXmlDeclaration.
-                settings = new XmlWriterSettings();
-                settings.OmitXmlDeclaration = true;
-
-                //To get rid of the default namespaces we create a new
-                //set of namespaces with one empty entry.
-                ns = new XmlSerializerNamespaces();
-                ns.Add("", "");
-
-                StringBuilder sb = new StringBuilder();
-
-                xs = new XmlSerializer(type);
+            var xs = new XmlSerializer(type);
+            var sb = new StringBuilder();
 
-                //We create a new XmlWriter with the previously created settings 
-                //(to OmitXmlDeclaration).
-                xw = XmlWriter.Create(sb, settings);
+            //To get rid of the default namespaces we create a new
+            //set of namespaces with one empty entry.
+            var ns = new XmlSerializerNamespaces();
+            ns.Add("", "");
 
-                //We call xs.Serialize and pass in our custom 
-                //XmlSerializerNamespaces object.
-                xs.Serialize(xw, bill, ns);
+            //To get rid of the xml declaration we create an 
+            //XmlWriterSettings object and tell it to OmitXmlDeclaration.
+            var settings = new XmlWriterSettings() { OmitXmlDeclaration = true };
 
-                xw.Flush();
-
-                outString = sb.ToString();
-            }
-            catch (Exception ex)
+            //We create a new XmlWriter with the previously created settings 
+            //(to OmitXmlDeclaration).
+            using (var xw = XmlWriter.Create(sb, settings))
             {
-                Console.WriteLine(ex.ToString());
-            }
-            finally
-            {
-                if (xw != null)
+                try
+                {
+                    //We call xs.Serialize and pass in our custom 
+                    //XmlSerializerNamespaces object.
+                    xs.Serialize(xw, bill, ns);
+                    xw.Flush();
+                }
+                catch (Exception ex)
                 {
-                    xw.Close();
+                    Console.WriteLine(ex.ToString());
```

Comment:
Instead of logging the error in Console, can we log it to Log4Net or somewhere else to review the exception later?

Reply:
The same as above, i didn't want to modify the component to allow logging